### PR TITLE
[ENH] Refactor X_inner_mtype

### PR DESCRIPTION
**Pull request by TonyBagnall**
part of #42 dependent on #838 

Refactors "x_inner_mtype" tag to be "x_inner_type"

We now have

input_type/output_type: Abstract data type one of Series, Collection or Tabular.
y_inner_type, X_inner_type: concrete data structure implementing ADT for internal data storage

no need to reinvent the concept of an abstract data type and confuse every one with scitype/mtypes.

### CODE REVIEW

**Review by MatthewMiddlehurst — COMMENTED** [review]
just noticed these are duplicate tags, dont have to edit here

```
    (
        "non-deterministic",
        "estimator",
        "bool",
        "does running the estimator multiple times generate the same output?",
    ),
    (
        "cant-pickle",
        "estimator",
        "bool",
        "flag for estimators which are unable to be pickled",
    ),
```

File: aeon/registry/_tags.py [hunk 1]
```diff
@@ -141,7 +141,7 @@
         "which data structure is the internal _fit/_predict able to deal with?",
     ),
     (
-        "X_inner_mtype",
+        "x_inner_type",
```
Comment (MatthewMiddlehurst) [inline aeon/registry/_tags.py:144]: Should this be "X_inner_type" to match the variable name? May be confusing otherwise

Comment (TonyBagnall) [reply aeon/registry/_tags.py:144]: good spot. pycharm find and replace seems to be not case sensitive
image

**Comment by TonyBagnall** [comment]
> "does running the estimator multiple times generate the same output?"

**Comment by TonyBagnall** [comment]
> just noticed there is are duplicate tags, dont have to edit here
> 
> ```
>     (
>         "non-deterministic",
>         "estimator",
>         "bool",
>         "does running the estimator multiple times generate the same output?",
>     ),
>     (
>         "cant-pickle",
>         "estimator",
>         "bool",
>         "flag for estimators which are unable to be pickled",
>     ),
> ```

I cant find this, I havent changed these tags, this PR is literally a find/replace on X_inner_mtype

File: extension_templates/transformer_simple.py [hunk 2]
```diff
@@ -230,8 +230,8 @@ def _transform(self, X, y=None):
 
         Parameters
         ----------
-        X : Series or Panel of mtype X_inner_mtype
-            if X_inner_mtype is list, _transform must support all types in it
+        X : Series or Panel of mtype x_inner_type
+            if x_inner_type is list, _transform must support all types in it
```
Comment (MatthewMiddlehurst) [inline extension_templates/transformer_simple.py:234]: It is still lower case outside the main package i.e. here, same for docs and stuff.

Fine with approving after that

**Pull request merged**